### PR TITLE
Add prefetch rows tweakable

### LINKED ISSUE #39: Request for setting statement properties ROW_PREFETCH and LOBPREFETCH_SIZE
I would like to be able to set the properties DEFAULT_LOBPREFETCH_SIZE and DEFAULT_ROW_PREFETCH when connections are created (incl. pooled connections from r2d2_oracle): https://docs.oracle.com/cd/E18283_01/appdev.112/e10646/ociaahan.htm#sthref5292

### PR DESCRIPTION
**Pull request by daviessm**
Partially fixes #39

### CODE REVIEW

File: src/connection.rs [hunk 1]
```diff
@@ -613,6 +613,24 @@ impl Connection {
     /// # Ok::<(), Error>(())
     /// ```
     ///
+    /// By default, a maximum of 2 rows are returned when the query is first
+    /// executed. To modify this, use `StmtParam::PrefetchRows(u32)` to customize
+    /// it. For more information on the difference between this and `FetchArraySize`,
+    /// see [this writeup](https://blog.dbi-services.com/arraysize-or-rowprefetch-in-sqlplus/).
```
Comment (cjbj) [inline src/connection.rs:619]: Another reference is https://oracle.github.io/node-oracledb/doc/api.html#rowfetching

File: src/statement.rs [hunk 2]
```diff
@@ -59,6 +59,17 @@ pub enum StmtParam {
     /// `StmtParam::FetchArraySize(1)`.
     FetchArraySize(u32),
 
+    /// The number of rows that will be prefetched by the Oracle Client
+    /// ibrary when a query is executed. The default value is
```
Comment (cjbj) [inline src/statement.rs:63]: Typo: ibrary

**Comment by kubo** [comment]
@daviessm Thanks for the PR and sorry for not replying so long.

I don't merge this PR because the new variant `PrefetchRows` causes a breaking change. I added a new method [`Connection.statement()`](https://github.com/kubo/rust-oracle/blob/acc865e63daabfe8ff1119627a79569b95a5bf86/src/connection.rs#L564-L621), which doesn't cause a breaking change when a new parameter is added. Could you make a new pull request which adds the `prefetch_rows()` method to [`StatementBuilder`](https://github.com/kubo/rust-oracle/blob/acc865e63daabfe8ff1119627a79569b95a5bf86/src/statement.rs#L47-L113)? Otherwise, may I do it based on your PR?

**Comment by daviessm** [comment]
I'm happy either way - if it's easier for you to do it yourself, go for it!

**Comment by kubo** [comment]
I did it by 7ec5577, which closes this PR.

**Pull request closed without merging**